tracer/test/test-applications/integrations/Samples.AWS.SQS/AsyncHelpers.cs

```diff
@@ -44,11 +72,77 @@ public static async Task SendAndReceiveMessages(AmazonSQSClient sqsClient)
                 _messsageBatchBarrier.SignalAndWait(); // Start the BatchedMessage receive loop
                 await receiveMessageBatchTask;
 
-                await DeleteQueueAsync(sqsClient);
+                await DeleteQueuesAsync(sqsClient);
             }
         }
 
-        private static async Task CreateSqsQueueAsync(AmazonSQSClient sqsClient)
+        [Flags]
+        public enum Scenario
+        {
+            None = 0,
+            Batch = 1,
+            InThread = 2,
+            Injected = 4
+        }
+
+        public static async Task RunSpecificScenario(AmazonSQSClient sqsClient, Scenario s)
+        {
+            // setup
+            await CreateSqsQueuesAsync(sqsClient);
+            await GetQueuesUrlAsync(sqsClient);
+            await PurgeQueueAsync(sqsClient);
+
+            Console.WriteLine($"Running test scenario {s}");
+            if (!s.HasFlag(Scenario.InThread))
+            {
+                if (s.HasFlag(Scenario.Injected))
+                {
+                    if (s.HasFlag(Scenario.Batch))
+                    {
+                        await SendBatchMessagesWithInjectedHeadersAsync(sqsClient);
+                    }
+                    else
+                    {
+                        await SendMessagesWithInjectedHeadersAsync(sqsClient);
+                    }
+                }
+                else
+                {
+                    if (s.HasFlag(Scenario.Batch))
+                    {
+                        await SendBatchMessagesWithoutInjectedHeadersAsync(sqsClient);
+                    }
+                    else
+                    {
+                        await SendMessagesWithoutInjectedHeadersAsync(sqsClient);
+                    }
+                }
+            }
+            else
+            {
+                // Note: there is no non-injected in-thread scenario
```

Comment:
The test _is_ running one though, so we should probably avoid that combination in the test and throw an exception if it's run here. Otherwise it could be a bit confusing

Reply:
hmm, ok. What happens is that the same test runs twice (since the param value is ignored in that config). But I could exclude it yes, that may be clearer.